pkg/library/libraryapplyconfiguration/operator_launch_helpers.go

```diff
@@ -153,10 +153,23 @@ func AdaptRunFn(fn func(ctx context.Context, workers int)) RunFunc {
 	}
 }
 
-func AdaptSyncFn(eventRecorder events.Recorder, controllerName string, originalRunOnce func(ctx context.Context, syncCtx factory.SyncContext) error) NamedRunOnce {
-	return NewNamedRunOnce(controllerName, func(ctx context.Context) error {
+type Syncer interface {
+	Sync(ctx context.Context, controllerContext factory.SyncContext) error
+}
+
+type ControllerWithInstanceName interface {
+	ControllerInstanceName() string
+}
+
+func AdaptSyncFn(eventRecorder events.Recorder, controller Syncer) NamedRunOnce {
```

Comment:
keep the old `AdaptSyncFn` and add `AdaptNamedController` that does this panic.

Reply:
why not, it will make the transition easier and we could actually merge this pr sooner.